### PR TITLE
Add Clang-Format File

### PR DESCRIPTION
**Pull request by chandramouli6890**
# Why

code formatting is essential while collaborating on open-source code spaces. This has several advantages

* consistently formatted code is easy to read
* multi-edits over the entire projects becomes easier due to consistent formatting and coding style
* PRs will not have unwanted whitespace changes, as devs tend to have different style rules configured in their text editor

# How

~I have copied the `.clang-format` file form [here](https://github.com/kehanXue/google-style-clang-format). This file contains the style followed in Google.~

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+CompactNamespaces: false
+ContinuationIndentWidth: 4
+Cpp11BracedListStyle: true
+DerivePointerAlignment: false  # Make sure the * or & align on the left
+EmptyLineBeforeAccessModifier: LogicalBlock
+FixNamespaceComments: true
+IncludeBlocks: Preserve
+IndentCaseLabels: true
+IndentPPDirectives: None
+IndentWidth: 2
```
Comment (modjtabaf) [inline .clang-format:55]: I prefer an indent width of 4. Any objections?

**Comment by modjtabaf** [comment]
This is a great addition to pooya. Thanks for adding it. IMO a few minor customizations are needed before this can be merged:
1 - an indent width of 4 is preferred
2 - it is better to avoid braces for single-line pieces of code

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  AfterStruct: false
+  AfterControlStatement: Never
+  AfterEnum: false
+  AfterFunction: false
+  AfterNamespace: false
+  AfterUnion: false
+  AfterExternBlock: false
+  BeforeCatch: false
+  BeforeElse: false
+  BeforeLambdaBody: false
```
Comment (modjtabaf) [inline .clang-format:36]: Please change these to `true`

Comment (chandramouli6890) [reply .clang-format]: done!

**Review by modjtabaf — APPROVED** [review]

**Pull request merged**